reana/reana_dev/git.py

```diff
@@ -815,4 +828,58 @@ def _commit_and_publish_version_bumps(components):
         )
 
 
+@click.option(
+    "--component",
+    "-c",
+    multiple=True,
+    default=["CLUSTER"],
+    help="Which components? [shortname|name|.|CLUSTER|ALL]",
+)
+@git_commands.command(name="git-create-pr")
+def git_create_pr_command(component):  # noqa: D301
+    """Creates a GitHub pull request for each selected component.
+
+    \b
+    :param components: The option ``component`` can be repeated. The value may
+                       consist of:
+                         * (1) standard component name such as
+                               'reana-workflow-controller';
+                         * (2) short component name such as 'r-w-controller';
+                         * (3) special value '.' indicating component of the
+                               current working directory;
+                         * (4) special value 'CLUSTER' that will expand to
+                               cover all REANA cluster components [default];
+                         * (5) special value 'CLIENT' that will expand to
+                               cover all REANA client components;
+                         * (6) special value 'DEMO' that will expand
+                               to include several runable REANA demo examples;
+                         * (7) special value 'ALL' that will expand to include
+                               all REANA repositories.
+    :type component: str
+    """
+
+    def _git_create_pr(comp):
+        """Create a pull request for the provided component."""
+        for cmd in [
+            "git push origin HEAD",
+            "hub pull-request -p --no-edit",
+            "hub pr list -L 1",
+        ]:
+            run_command(cmd, component)
+
+    for component in select_components(component):
+        if (
+            get_current_branch(get_srcdir(component)) == "master"
+        ):  # replace with is_feature_branch from #371
+            display_message(
+                "You are trying to create PR but the current branch is master, please amend.",
+                component,
+            )
+        else:
+            report_component_branch_difference(
+                component, "upstream/master", exit_code=True
+            )
```

Comment:
Please add some informative message to people, currently the would see:

```console
$ reana-dev git-create-pr -c r-w-controller
reana-workflow-controller @ test2 (1 AHEAD 5 BEHIND upstream/master) @ 4faf564 test
```

It may be good to say something like:

```
Error: please rebase your feature branch against latest master.
```

(Although beware that there will be `maint-0.6` and `maint-0.7`, but we shall upgrade the command later to account for this possibility (=to check against a different branch than master). For now, the simple error message is enough.)